crates/ide/src/inlay_hints.rs

```diff
@@ -813,7 +813,8 @@ fn ty_to_text_edit(
     config: &InlayHintsConfig,
     node_for_hint: &SyntaxNode,
     ty: &hir::Type,
-    offset_to_insert: TextSize,
+    offset_to_insert_ty: TextSize,
+    range_to_insert_braces: Option<TextRange>,
```

Comment:
```suggestion
    additional_edits: &dyn Fn(&mut TextEditBuilder)
```
Let's make this take a callback instead of something so specify